src/key_values/error.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt;
+
+/// An error encountered while working with structured data.
+#[derive(Clone, Debug)]
+pub struct KeyValueError {
```

Comment:
An alternative here could be to make this error like `fmt::Error`, so it doesn't carry any context on its own and is easy to create.